doc/changelog/11-standard-library/18538-deprecate_Bool_nat.rst

```diff
@@ -0,0 +1,6 @@
+- **Deprecated:**
+  The library file ``Coq.Arith.Bool_nat`` has been deprecated.
+  Users who think this file is of general interest are encouraged to open an
+  issue to explain why it is useful to them.
```

Comment:
We probably don't need that sentence. The file seems mostly duplication for historical reasons.